x-pack/plugins/code/server/utils/with_request.ts

```diff
@@ -11,8 +11,17 @@ export class WithRequest {
   public readonly callWithRequest: (endpoint: string, clientOptions?: AnyObject) => Promise<any>;
 
   constructor(readonly req: Request) {
-    this.callWithRequest = req.server.plugins.elasticsearch
-      .getCluster('data')
-      .callWithRequest.bind(null, req);
+    const cluster = req.server.plugins.elasticsearch.getCluster('data');
+
+    // @ts-ignore
+    const securityPlugin = req.server.plugins.security;
+    if (securityPlugin) {
+      const useRbac = securityPlugin.authorization.mode.useRbacForRequest(req);
+      if (useRbac) {
+        this.callWithRequest = cluster.callWithInternalUser;
```

Comment:
nit: this variable name is misleading. `callWithRequest` is only assigned `cluster.callWithRequest` on certain occasions, but on others, it's assigned `callWithInternalUser`. Maybe something more generic like `callCluster` would be a more appropriate name?

Reply:
done